add-on/panels/test/panel_test.js

```diff
@@ -1561,4 +1565,124 @@ describe("loop.panel", function() {
+  describe("SharePanelView", function() {
+    var view, dispatcher, roomStore;
+
+    function createTestComponent(extraProps) {
+      var props = _.extend({
+        dispatcher: dispatcher,
+        store: roomStore
+      }, extraProps);
+      return TestUtils.renderIntoDocument(
+        React.createElement(loop.panel.SharePanelView, props));
+    }
+
+    beforeEach(function() {
+      dispatcher = new loop.Dispatcher();
+      sandbox.stub(dispatcher, "dispatch");
+      roomStore = new loop.store.RoomStore(dispatcher, {
+        constants: {}
+      });
+
+      roomStore.setStoreState({
+        activeRoom: {
+          roomToken: "fakeToken"
+        },
+        openedRoom: null,
+        pendingCreation: false,
+        pendingInitialRetrieval: false,
+        rooms: [],
+        error: undefined
+      });
+
+      view = createTestComponent();
+      sandbox.stub(view, "closeWindow");
+    });
+
+    it("should not open the panel if there is no room pending of creation", function() {
+      expect(view.getDOMNode()).eql(null);
+    });
+
+    it("should open the panel after room creation", function() {
+      var clock = sinon.useFakeTimers();
+      // Simulate that the user has click the browse button
+      roomStore.setStoreState({
+        pendingCreation: true
+      });
+
+      // Room has been created succesfully
+      roomStore.setStoreState({
+        pendingCreation: false
+      });
+
+      var panel = view.getDOMNode();
+      clock.tick(loop.panel.SharePanelView.SHOW_PANEL_DELAY);
+
+      expect(view.state.showPanel).eql(true);
+      expect(panel.classList.contains("share-panel-open")).eql(true);
+    });
+
+    it("should close the share panel when clicking the overlay", function() {
+      view.setState({
+        showPanel: true
+      });
+
+      var overlay = view.getDOMNode().querySelector(".share-panel-overlay");
+      var panel = view.getDOMNode();
+
+      TestUtils.Simulate.click(overlay);
+
+      expect(view.state.showPanel).eql(false);
+      expect(panel.classList.contains("share-panel-open")).eql(false);
+    });
+
+    it("should close the share panel when clicking outside the panel", function() {
+      view.setState({
+        showPanel: true
+      });
+
+      var panel = view.getDOMNode();
+
+      view._onDocumentVisibilityChanged({
+        target: {
+          hidden: true
+        }
+      });
+
+      expect(view.state.showPanel).eql(false);
+      expect(panel.classList.contains("share-panel-open")).eql(false);
+    });
+
+    it("should close the hello panel when clicking outside the panel", function() {
+      view.setState({
+        showPanel: true
+      });
+
+      view._onDocumentVisibilityChanged({
+        target: {
+          hidden: true
+        }
+      });
+
+      expect(view.state.showPanel).eql(false);
+      sinon.assert.calledOnce(view.closeWindow);
+    });
+
+    describe("#openRoom", function() {
+      it("should open the room and close the panel", function() {
```

Comment:
This should be part of the main section where we're testing closing the panel. Something like change the visibility & check that `openRoom` is called - note that this doesn't (and shouldn't) test the close panel any more.